varsim.py

```diff
@@ -21,6 +24,51 @@
 REQUIRE_VARSIMJAR = not os.path.isfile(DEFAULT_VARSIMJAR)
 if REQUIRE_VARSIMJAR: DEFAULT_VARSIMJAR = None
 
+def convertCN(filenames, operation):
+    """
+    convert '2/1'-like copy number to a single number(e.g. 2)
+    0 will be considered same as 1
+    by default the max number will be kept
+    the change is in place
+    """
+    if operation != "two2one" and operation != "one2two":
+        raise ValueError("Only two2one or one2two allowed")
+    two2one = operation == "two2one"
+    for name in filenames:
+        file = open(name, "r")
+        output = tempfile.NamedTemporaryFile(mode = 'r+w', delete = False)
+        for l in file:
+            l = string.rstrip(l)
+            fields = string.split(l, "\t")
+            if string.find(l,"#") == 0 or string.find(fields[8],"CN") < 0:
+                output.write(l + "\n")
+            else:
+                info = string.split(fields[8], ':')
+                cnIndex = info.index('CN')
+                gtIndex = info.index('GT')
+                #change CN field in all samples
+                for sampleIndex in range(9,len(fields)):
+                    sampleInfo = string.split(fields[sampleIndex], ':')
+                    if two2one:
+                        cn = re.compile('[/|]').split(sampleInfo[cnIndex])
```

Comment:
The pattern `re.compile("[/|]")` can be a constant compiled once instead of in every loop iteration.